### PR TITLE
doc: Remove upgrade note in release notes from EOL versions

### PR DESCRIPTION
**Pull request by maflcko**
Remove the bloat on how to upgrade from version that are EOL

### CODE REVIEW

**Comment by MarcoFalke** [comment]
(Can be reviewed as GitHub rich diff)

**Comment by laanwj** [comment]
IMO, the underline syntax for headers is easier to read when reading the text instead of formatted version, this is, as far as i know, why we've always stuck with that.

ACK on removing unnecessary downgrading instructions.

**Comment by MarcoFalke** [comment]
Restored

File: doc/release-notes.md [hunk 1]
```diff
@@ -24,14 +24,9 @@ shut down (which might take a few minutes for older versions), then run the
...
 
-The first time you run version 0.15.0, your chainstate database will be converted to a
-new format, which will take anywhere from a few minutes to half an hour,
-depending on the speed of your machine.
-
-Note that the block database format also changed in version 0.8.0 and there is no
-automatic upgrade code from before version 0.8 to version 0.15.0. Upgrading
-directly from 0.7.x and earlier without redownloading the blockchain is not supported.
-However, as usual, old wallet versions are still supported.
+Upgrading directly from a version of Bitcoin Core that reached its EOL is
```
Comment (fanquake) [inline doc/release-notes.md]: nit :s/that reached its/that has reached its

File: doc/release-notes.md [hunk 2]
```diff
@@ -24,14 +24,9 @@ shut down (which might take a few minutes for older versions), then run the
...
+possible, but might take some time in case the datadir needs to be migrated in
```
Comment (fanquake) [inline doc/release-notes.md]: nit: s/some time in case the datadir needs to be migrated in the background./some time if the datadir needs to be migrated.

**Review by fanquake — APPROVED** [review]
utACK faab858

**Comment by fanquake** [comment]
re-utACK fa346fe

**Comment by luke-jr** [comment]
utACK

**Pull request merged**